File: php/classes/Vaga.php
<?php

require_once __DIR__ . "/enum/StatusVagas.php";

class Vaga
{
    private int $empresa_id;
    private string $titulo;
    private string $descricao;
    private string $area;
    private StatusVaga $status;

    public function __construct(int $empresa_id, string $titulo, string $descricao, string $area, StatusVaga $status = StatusVaga::ABERTA)
    {
        $this->empresa_id = $empresa_id;
        $this->titulo = $titulo;
        $this->descricao = $descricao;
        $this->area = $area;
        $this->status = $status;
    }

    public function getEmpresaId(): int
    {
        return $this->empresa_id;
    }

    public function setEmpresaId(int $empresa_id): void
    {
        $this->empresa_id = $empresa_id;
    }

    public function getTitulo(): string
    {
        return $this->titulo;
    }

    public function setTitulo(string $titulo): void
    {
        $this->titulo = $titulo;
    }

    public function getDescricao(): string
    {
        return $this->descricao;
    }

    public function setDescricao(string $descricao): void
    {
        $this->descricao = $descricao;
    }

    public function getArea(): string
    {
        return $this->area;
    }

    public function setArea(string $area): void
    {
        $this->area = $area;
    }

    public function getStatus(): StatusVaga
    {
        return $this->status;
    }

    public function setStatus(StatusVaga $status): void
    {
        $this->status = $status;
    }

}

File: java/src/main/java/model/Vaga.java
package model;

import java.time.LocalDateTime;

public class Vaga extends EntidadeBase {

    private Integer empresaId;
    private String titulo;
    private String descricao;
    private String area;
    private String status;
    private LocalDateTime criadaEm;

    public Vaga() { super(); }

    public Vaga(Integer id, Integer empresaId, String titulo,
                String descricao, String area, String status, LocalDateTime criadaEm) {
        super(id);
        this.empresaId = empresaId;
        this.titulo    = titulo;
        this.descricao = descricao;
        this.area      = area;
        this.status    = status;
        this.criadaEm  = criadaEm;
    }

    public Integer getEmpresaId() { return empresaId; }
    public void setEmpresaId(Integer empresaId) { this.empresaId = empresaId; }

    public String getTitulo()   { return titulo; }
    public void setTitulo(String titulo) { this.titulo = titulo; }

    public String getDescricao() { return descricao; }
    public void setDescricao(String descricao) { this.descricao = descricao; }

    public String getArea()     { return area; }
    public void setArea(String area) { this.area = area; }

    public String getStatus()   { return status; }
    public void setStatus(String status) { this.status = status; }

    public LocalDateTime getCriadaEm() { return criadaEm; }
    public void setCriadaEm(LocalDateTime criadaEm) { this.criadaEm = criadaEm; }

    @Override
    public String toString() { return titulo + " [" + status + "]"; }
}